packages/compass-query-bar/src/utils/query-storage.ts

```diff
@@ -2,41 +2,66 @@ import { UUID, EJSON } from 'bson';
 import { orderBy } from 'lodash';
 import { type BaseQuery } from '../constants/query-properties';
 import { UserData } from '@mongodb-js/compass-user-data';
+import { z } from 'zod';
 
 // We do not save maxTimeMS
-export type RecentQuery = Omit<BaseQuery, 'maxTimeMS'> & {
-  _id: string;
-  _lastExecuted: Date;
-  _ns: string;
-  _host?: string;
-};
+const BaseQuerySchema: z.Schema<Omit<BaseQuery, 'maxTimeMS'>> = z.object({
+  filter: z.any().optional(),
+  project: z.any().optional(),
+  collation: z.any().optional(),
+  sort: z.any().optional(),
+  skip: z.number().optional(),
+  limit: z.number().optional(),
+});
 
-export type FavoriteQuery = RecentQuery & {
-  _name: string;
-  _dateModified: Date;
-  _dateSaved: Date;
-};
+const RecentQuerySchema = BaseQuerySchema.and(
+  z.object({
+    _id: z.string().uuid(),
+    _lastExecuted: z
+      .union([z.coerce.date(), z.number()])
+      .transform((x) => new Date(x)),
+    _ns: z.string(),
```

Comment:
I did add fixtures from v1.27.0 and it is there. Do i need to go more back to support this? I can make this optional, but just make it clear it has been there since 2017.

Reply:
No, I don’t think we need to go further back than that :wink:  My question wasn’t so much when this was introduced as whether a previous Compass version would have errored here vs. not